GeorgianMemenator/Scenes/DogBreedDetail/DogBreedDetailViewModel.swift

```diff
@@ -4,9 +4,37 @@
 //
 //  Created by Qeti Avanashvili on 22.03.23.
 //
-
 import Foundation
+import Combine
 
 class DogBreedDetailViewModel: ObservableObject {
+    @Published var dogBreedDetail: DogBreed?
+    @Published private(set) var hasError: Bool = false
+    @Published private(set) var isLoading: Bool = false
+    
+    private let dogBreedDetailService: DogBreedsServing
+    private let dogBreedId: String
+    private var cancellables = Set<AnyCancellable>()
     
+    init(dogBreedDetailService: DogBreedsServing, dogBreedId: String) {
+        self.dogBreedDetailService = dogBreedDetailService
+        self.dogBreedId = dogBreedId
+    }
+   
+    func getDogBreedDetail() {
+        dogBreedDetailService
+            .getDogBreedDetails(with: dogBreedId)
+            .sink(
+                receiveCompletion: { [weak self] val in
+                    if case .failure = val {
+                        self?.isLoading = false
```

Comment:
Once you remove isLoading and hasError, you will need to delete these lines

Reply:
Good question! since you have completion block empty, Xcode says that you are not using `weak self` at all. So you can remove it for now.

Reply:
<img width="1328" alt="Screenshot 2023-04-03 at 14 16 44" src="https://user-images.githubusercontent.com/111251425/229481536-766840ee-d5b2-4eab-aab0-40984be15107.png">
 I have fixed it.